From 3a4be68986691c7a443d27163507dc6b31a87fe1 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:04:26 -0400
Subject: [kernel] perf: Validate locking assumption

Message-id: <1410181752-25631-22-git-send-email-jolsa@redhat.com>
Patchwork-id: 91707
O-Subject: [PATCH RHEL7.1 BZ1134356 021/307] perf: Validate locking assumption
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 63342411efd2d9350ad405205da036cd45ed1640
Author: Peter Zijlstra <peterz@infradead.org>
Date: Mon May 5 11:49:16 2014 +0200

description
===========
Document and validate the locking assumption of event_sched_in().
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/events/core.c b/kernel/events/core.c
index 04fce87..d8b4639 100644
--- a/kernel/events/core.c
+++ b/kernel/events/core.c
@@ -1690,6 +1690,8 @@ event_sched_in(struct perf_event *event,
  u64 tstamp = perf_event_time(event);
  int ret = 0;
 
+ lockdep_assert_held(&ctx->lock);
+
  if (event->state <= PERF_EVENT_STATE_OFF)
   return 0;
 
-- 
1.7.1